metadata-load/src/main/java/uk/ac/ebi/ampt2d/metadata/importer/objectImporters/SraObjectsImporter.java

```diff
@@ -0,0 +1,139 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package uk.ac.ebi.ampt2d.metadata.importer.objectImporters;
+
+import org.springframework.core.convert.converter.Converter;
+import uk.ac.ebi.ampt2d.metadata.importer.SraRetrieverByAccession;
+import uk.ac.ebi.ampt2d.metadata.importer.objects.AnalysisObject;
+import uk.ac.ebi.ampt2d.metadata.importer.objects.ReferenceSequenceObject;
+import uk.ac.ebi.ampt2d.metadata.importer.objects.SampleObject;
+import uk.ac.ebi.ampt2d.metadata.importer.objects.StudyObject;
+import uk.ac.ebi.ampt2d.metadata.importer.xml.SraXmlParser;
+import uk.ac.ebi.ampt2d.metadata.persistence.entities.Study;
+import uk.ac.ebi.ena.sra.xml.LinkType;
+import uk.ac.ebi.ena.sra.xml.StudyType;
+import uk.ac.ebi.ena.sra.xml.XRefType;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Stream;
+
+public class SraObjectsImporter implements ObjectsImporter {
+
+    private static final String STUDY_QUERY = "SELECT STUDY_XML FROM ERA.STUDY WHERE STUDY_ID = :accession";
+
+    private static final Logger IMPORT_LOGGER = Logger.getLogger(SraObjectsImporter.class.getName());
+
+    private SraRetrieverByAccession sraRetrieverByAccession;
+
+    private SraXmlParser<StudyType> sraStudyXmlParser;
+
+    private Converter<StudyType, Study> studyConverter;
+
+    public SraObjectsImporter(SraRetrieverByAccession sraRetrieverByAccession,
+                              SraXmlParser<StudyType> sraStudyXmlParser,
+                              Converter<StudyType, Study> studyConverter) {
+        this.sraRetrieverByAccession = sraRetrieverByAccession;
+        this.sraStudyXmlParser = sraStudyXmlParser;
+        this.studyConverter = studyConverter;
+    }
+
+    @Override
+    public StudyObject importObject(StudyObject studyObject) {
+
+        Set<String> accessions = studyObject.getAccessions();
+
+        for (String accession : accessions) {
+            try {
+                String xml = sraRetrieverByAccession.getXml(accession);
+                StudyType studyType = sraStudyXmlParser.parseXml(xml, accession);
+                Study study = studyConverter.convert(studyType);
+                AnalysisObject analysisObject = new AnalysisObject(getAnalysisAccessions(studyType), new ArrayList<>());
+                analysisObject.accept(this);
```

Comment:
After going through the code flow, I can only see the `XXXObject` classes as another indirection level, but without providing a clear functionality. If they were removed, the `importObject` methods could be renamed as `importStudy`, `importAnalysis`, etc, and receive the accession(s) as an argument.

Reply:
I couldn't use the metadata "study" object directly because the visitor pattern needs an object with a state to visit and process using it.  I can use a static method but it doesn't look a cleaner way as we need to pass both importer and accessions manually instead of using the state(accessions) of the object, and also static methods could not be overloaded.

Reply:
I can use like below in Study but am not using the state of Study, instead manually sending the accessions and the static method cannot be overridden,
```
 public static List<Study> accept(ObjectsImporter objectsImporter, Set<String> accessions) {
        return objectsImporter.importStudy(accessions);
    }
```
is it ok ?

Reply:
have implemented calling the SraObjectImporter respective importXXX methods.